examples/gno.land/r/demo/profile/profile_test.gno

```diff
@@ -0,0 +1,157 @@
+package profile
+
+import (
+	"std"
+	"testing"
+
+	"gno.land/p/demo/testutils"
+)
+
+const (
+	DisplayName        = "DisplayName"
+	Homepage           = "Homepage"
+	Bio                = "Bio"
+	Age                = "Age"
+	AvailableForHiring = "AvailableForHiring"
+	InvalidField       = "InvalidField"
+)
+
+func TestStringFields(t *testing.T) {
+	addr := testutils.TestAddress("alice")
+	std.TestSetRealm(std.NewUserRealm(addr))
+
+	// Get before setting
+	name := GetStringField(addr, DisplayName, "anon")
+	assertSameString(t, name, "anon")
+
+	// Set
+	if err := SetStringField(DisplayName, "Alice foo"); err != nil {
+		t.Errorf("unexpected error: %v", err)
+	}
+	if err := SetStringField(Homepage, "https://example.com"); err != nil {
+		t.Errorf("unexpected error: %v", err)
+	}
+
+	// Get after setting
+	name = GetStringField(addr, DisplayName, "anon")
+	homepage := GetStringField(addr, Homepage, "")
+	bio := GetStringField(addr, Bio, "42")
+
+	assertSameString(t, name, "Alice foo")
+	assertSameString(t, homepage, "https://example.com")
+	assertSameString(t, bio, "42")
+}
+
+func TestIntFields(t *testing.T) {
+	addr := testutils.TestAddress("bob")
+	std.TestSetRealm(std.NewUserRealm(addr))
+
+	// Get before setting
+	age := GetIntField(addr, Age, 25)
+	assertSameInt(t, age, 25)
+
+	// Set
+	if err := SetIntField(Age, 30); err != nil {
+		t.Errorf("unexpected error: %v", err)
+	}
+
+	// Get after setting
+	age = GetIntField(addr, Age, 25)
+	assertSameInt(t, age, 30)
```

Comment:
and the new `uassert.Equal`

Reply:
https://github.com/gnolang/gno/pull/1983/commits/5789de857dbe5122bcf16bc5eaeaefd5d779a57c